src/utils/hooks/useLoadTemplateNEW.tsx

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useState } from 'react'
+import {
+  GetTemplateQuery,
+  Template,
+  TemplateSection,
+  useGetTemplateQuery,
+} from '../generated/graphql'
+import { getTemplateSections } from '../helpers/application/getSectionsDetails'
+import { SectionDetails, TemplateDetails } from '../types'
+
+interface useLoadTemplateProps {
+  templateCode?: string
+}
+
+const useLoadTemplate = ({ templateCode }: useLoadTemplateProps) => {
+  const [template, setTemplate] = useState<TemplateDetails>()
+  const [sections, setSectionsDetails] = useState<SectionDetails[]>()
+  const [elementsIds, setElementsIds] = useState<number[]>([])
+  const [error, setError] = useState('')
+
+  const { data, loading: apolloLoading, error: apolloError } = useGetTemplateQuery({
+    variables: {
+      code: templateCode || '',
+    },
+    skip: !templateCode,
+  })
+
+  useEffect(() => {
+    if (!data) return
+
+    // Check that only one template matched
+    let error = checkForTemplateErrors(data)
+    if (error) {
+      setError(error)
+      return
+    }
```

Comment:
Ah, I think the problem is that we need to return after finding some relation problem in the fetched data, because next lines will be depending on that. And can't use the state of `error` right away because it's not yet updated.

Reply:
Yeah okay, I see 👍